syncer/syncer.go

```diff
@@ -2205,6 +2205,8 @@ func (s *Syncer) Close() {
 	s.Lock()
 	defer s.Unlock()
 
+	s.removeLabelValuesWithTaskInMetrics(s.cfg.Name)
```

Comment:
How about putting it in a later position (before other parts already stopped, like `s.closed.Set(true)`)? if so, ditto for other `remove***` methods.

Reply:
done